### DISCUSSION TITLE
Push Notification

### DISCUSSION THREAD

**Discussion by Bhargav-IndiaNIC** [GetStream/stream-chat-android#4793] [General]
Hello

In My project, everything works properly but I did not get push notifications. I am also configuring a push notification provider in the dashboard but not getting push notifications. In the dashboard click on Test Push Configuration button and add Configuration and user id then click on the test notification button that time getting the result below but nor receive the push notification in the device.

{
    "rendered_message": {
        "type": "message.new",
        "cid": "messaging:!members-iN1YCGw2JXokFsA-524Qhas8uCe1maFy73c-g5wZHXs",
        "receiver_id": "dev_user_808",
        "sender": "stream.chat",
        "version": "v2",
        "message_id": "dev_user_885-9875134a-8309-41cb-b312-b7effb23251c",
        "id": "dev_user_885-9875134a-8309-41cb-b312-b7effb23251c",
        "channel_type": "messaging",
        "channel_id": "!members-iN1YCGw2JXokFsA-524Qhas8uCe1maFy73c-g5wZHXs"
    },
    "duration": "276.96ms"
}

**Comment by JcMinarro** [comment]
Could you share with us how you are configuring Push Notification on the ChatClient?
[Here](https://getstream.io/chat/docs/sdk/android/client/guides/push-notifications/setup/) you can find the Push Notification Docs Section that will guide you how to configure PN in your ChatClient

**Reply by Bhargav-IndiaNIC** [reply]
Hello @JcMinarro 

PushNotification configuring Code 

```
fun init(apiKey: String) {
        val notificationHandler = NotificationHandlerFactory.createNotificationHandler(
            context = context,
            newMessageIntent = {
                    messageId: String,
                    channelType: String,
                    channelId: String,
                ->
                Intent(context, HomeActivity::class.java).apply {
                    putExtra(EXTRA_CHANNEL_ID, channelId)
                    putExtra(EXTRA_CHANNEL_TYPE, channelType)
                    putExtra(EXTRA_MESSAGE_ID, messageId)
                }
            }
        )

        val offlinePlugin = StreamOfflinePluginFactory(
            Config(
                backgroundSyncEnabled = true,
                userPresence = true,
                persistenceEnabled = true,
                useSequentialEventHandler = true
            ),
            context
        )
        val notificationConfig =
            NotificationConfig(
                true,
                pushDeviceGenerators = listOf(
                    FirebasePushDeviceGenerator(providerName = PROVIDER_NAME)
                ),
                shouldShowNotificationOnPush = { true }
            )
        ChatClient.Builder(apiKey, context)
            .notifications(notificationConfig, notificationHandler)
            .logLevel(if (BuildConfig.DEBUG) ChatLogLevel.ALL else ChatLogLevel.NOTHING)
            .withPlugin(offlinePlugin)
            .debugRequests(true)
            .build()
    }
```
    
Above Init method call in MyApplication class

**Reply by lukicmarko** [reply]
I spent 20h of researching and then I figured out that I have to call the init method in  MyApplication class... I am curious why?

**Reply by JcMinarro** [reply]
Hello @lukicmarko 
The `ChatClient` needs to be initialized before the SDK handles the Push Notification to access the Stream Server and obtain all required data about the new message received.
The easiest way (If you don't want to implement your BroadcastReceiver) is initializing it into the Application class to ensure it is ready to be used when the SDK requires this data.
It is already detailed in the [documentation](https://getstream.io/chat/docs/sdk/android/client/guides/push-notifications/push-providers/firebase/)
